forc-pkg/src/manifest.rs

```diff
@@ -285,19 +285,29 @@ impl PackageManifestFile {
     /// standalone package.
     ///
     /// If this package is a member of a workspace, patches are fetched from
-    /// the workspace manifest file.
+    /// the workspace manifest file, ignoring any patch defined in the package
+    /// manifest file, even if a patch section is not defined in the namespace.
     pub fn resolve_patches(&self) -> Result<impl Iterator<Item = (String, PatchMap)>> {
-        let workspace_patches = self
-            .workspace()
-            .ok()
-            .flatten()
-            .and_then(|workspace| workspace.patch.clone());
-        let package_patches = self.patch.clone();
-        match (workspace_patches, package_patches) {
-            (Some(_), Some(_)) => bail!("Found [patch] table both in workspace and member package's manifest file. Consider removing [patch] table from package's manifest file."),
-            (Some(workspace_patches), None) => Ok(workspace_patches.into_iter()),
-            (None, Some(pkg_patches)) => Ok(pkg_patches.into_iter()),
-            (None, None) => Ok(BTreeMap::default().into_iter()),
+        if let Some(workspace) = self.workspace().ok().flatten() {
```

Comment:
```suggestion
       if let Some(workspace) = self.workspace().ok().flatten() {
            // If workspace is defined, passing a local patch is a warning, but the global patch is used
            if self.patch.is_some() {
                println_warning(
                    "patch for the non root package will be ignored, specify patch at the workspace root:"
                );
                println_warning(&format!(
                    "package: {}",
                    workspace.path().to_str().unwrap_or_default()
                ));
            }
            return Ok(workspace.patch.as_ref().cloned().unwrap_or_default().into_iter());
        }
        Ok(self.patch.as_ref().cloned().unwrap_or_default().into_iter())
```

I think this is a little cleaner.